jupyterlab/tests/test_app.py

```diff
@@ -174,6 +175,68 @@ def _process_finished(self, future):
             os._exit(1)
 
 
+jest_aliases = dict(base_aliases)
+jest_aliases.update({
+    'coverage': 'JestApp.coverage',
+    'pattern': 'JestApp.testPathPattern',
+    'watchAll': 'JestApp.watchAll'
+})
+
+
+class JestApp(ProcessTestApp):
+    """A notebook app that runs a jest test."""
+
+    coverage = Bool(False, help='Whether to run coverage').tag(config=True)
+
+    testPathPattern = Unicode('').tag(config=True)
+
+    watchAll = Bool(False).tag(config=True)
+
+    aliases = jest_aliases
+
+    jest_dir = Unicode('')
+
+    open_browser = False
+
+    def get_command(self):
+        """Get the command to run"""
+        terminalsAvailable = self.web_app.settings['terminals_available']
+        jest = './node_modules/.bin/jest'
+        debug = self.log.level == logging.DEBUG
+
+        if self.coverage:
+            cmd = [jest, '--coverage']
+        elif debug:
+            cmd = 'node --inspect-brk %s --no-cache' % jest
+            if self.watchAll:
+                cmd += ' --watchAll'
+            else:
+                cmd += ' --watch'
+            cmd = cmd.split()
+        else:
+            cmd = [jest]
+
+        if self.testPathPattern:
+            cmd += ['--testPathPattern', self.testPathPattern]
+
+        cmd += ['--runInBand']
+
+        config = dict(baseUrl=self.connection_url,
+                      terminalsAvailable=str(terminalsAvailable),
+                      token=self.token)
+
+        td = tempfile.mkdtemp()
+        atexit.register(lambda: shutil.rmtree(td, True))
+
+        config_path = os.path.join(td, 'config.json')
+        with open(config_path, 'w') as fid:
+            json.dump(config, fid)
+
+        env = os.environ.copy()
+        env['JUPYTER_CONFIG_DATA'] = config_path
+        return cmd, dict(cwd=self.jest_dir, env=env)
+
+
 class KarmaTestApp(ProcessTestApp):
```

Comment:
Can this be removed?

Reply:
Woops, I forgot to read the title of this issue :) I thought this replaced all tests, not just services tests.